src/eei.cpp

```diff
@@ -150,19 +151,22 @@ string toHex(evm_uint256be const& value) {
 
     if (import->base == Name("evmTrace")) {
       uint32_t pc = static_cast<uint32_t>(arguments[0].geti32());
-      uint32_t opcode = static_cast<uint32_t>(arguments[1].geti32());
+      int32_t opcode = arguments[1].geti32();
       uint32_t cost = static_cast<uint32_t>(arguments[2].geti32());
       int32_t sp = arguments[3].geti32();
 
       HERA_DEBUG << "evmTrace\n";
 
       static constexpr int stackItemSize = sizeof(evm_uint256be);
       heraAssert(sp <= (1024 * stackItemSize), "EVM stack pointer out of bounds.");
+      heraAssert(opcode >= 0x00 && opcode <= 0xff, "Invalid EVM instruction.");
+
+      string opName = evmInstructionNames.count(static_cast<uint8_t>(opcode)) ? evmInstructionNames.at(static_cast<uint8_t>(opcode)) : "UNKNOWN";
```

Comment:
You search twice here.

Reply:
`.at` would throw an exception the opcode is not found.

Reply:
What would the code look like? Is that any better / more concise?

Reply:
```
auto it = evmInstructionNames.find(static_cast<uint8_t>(opcode));
string opName = (it != emvInstructionNames.end()) ? it->second : "UNKNOWN";
```

Reply:
I always found the iterators not so readable, but it does seem to do less lookups this way.